target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts

```diff
@@ -594,6 +691,98 @@ export class PythSolanaReceiver {
+   */
+  async buildPostTwapUpdateInstructions(
+    twapUpdateDataArray: string[]
+  ): Promise<{
+    postInstructions: InstructionWithEphemeralSigners[];
+    priceFeedIdToTwapUpdateAccount: Record<string, PublicKey>;
+    closeInstructions: InstructionWithEphemeralSigners[];
+  }> {
+    const postInstructions: InstructionWithEphemeralSigners[] = [];
+    const priceFeedIdToTwapUpdateAccount: Record<string, PublicKey> = {};
+    const closeInstructions: InstructionWithEphemeralSigners[] = [];
+
+    const treasuryId = getRandomTreasuryId();
+
+    if (twapUpdateDataArray.length !== 2) {
+      throw new Error(
+        "twapUpdateDataArray must contain exactly two updates (start and end)"
+      );
+    }
+
+    const [startUpdateData, endUpdateData] = twapUpdateDataArray.map((data) =>
+      parseAccumulatorUpdateData(Buffer.from(data, "base64"))
+    );
+
+    // Validate that the start and end updates contain the same number of price feeds
+    if (startUpdateData.updates.length !== endUpdateData.updates.length) {
+      throw new Error(
+        "Start and end updates must contain the same number of price feeds"
+      );
+    }
+
+    // Post encoded VAAs
+    const {
+      postInstructions: buildVaasInstructions,
+      closeInstructions: closeVaasInstructions,
+      startVaa,
+      endVaa,
+    } = await buildPostEncodedVaasForTwapInstructions(
+      this.wormhole,
+      startUpdateData,
+      endUpdateData
+    );
+    postInstructions.push(...buildVaasInstructions);
+    closeInstructions.push(...closeVaasInstructions);
+
+    // Post a TWAP update to the receiver contract for each price feed
+    for (let i = 0; i < startUpdateData.updates.length; i++) {
+      const startUpdate = startUpdateData.updates[i];
+      const endUpdate = endUpdateData.updates[i];
+
+      const twapUpdateKeypair = new Keypair();
+      postInstructions.push({
+        instruction: await this.receiver.methods
+          .postTwapUpdate({
+            startMerklePriceUpdate: startUpdate,
+            endMerklePriceUpdate: endUpdate,
+            treasuryId,
+          })
+          .accounts({
+            startEncodedVaa: startVaa.encodedVaaKeypair.publicKey,
+            endEncodedVaa: endVaa.encodedVaaKeypair.publicKey,
+            twapUpdateAccount: twapUpdateKeypair.publicKey,
+            treasury: getTreasuryPda(treasuryId, this.receiver.programId),
+            config: getConfigPda(this.receiver.programId),
+          })
+          .instruction(),
+        signers: [twapUpdateKeypair],
+        computeUnits: POST_UPDATE_COMPUTE_BUDGET,
+      });
+
+      priceFeedIdToTwapUpdateAccount[
+        "0x" + parseTwapMessage(startUpdate.message).feedId.toString("hex")
+      ] = twapUpdateKeypair.publicKey;
+      closeInstructions.push(
+        await this.buildClosePriceUpdateInstruction(twapUpdateKeypair.publicKey)
```

Comment:
note: I think this doesn't work. This instructions only allow closing `PriceUpdateV2` s and not `TwapUpdate`

Reply:
Ooof good point looks like ill need to update the receiver's reclaim_rent method as well

Reply:
added a new `reclaimTwapRent` method to the recv contract